src/main/java/org/testng/ISuite.java

```diff
@@ -21,97 +21,89 @@
   /**
    * @return the name of this suite.
    */
-  public String getName();
+  String getName();
 
   /**
    * @return The results for this suite.
    */
-  public Map<String, ISuiteResult> getResults();
+  Map<String, ISuiteResult> getResults();
 
   /**
    * @return The object factory used to create all test instances.
    */
-  public IObjectFactory getObjectFactory();
-  public IObjectFactory2 getObjectFactory2();
+  IObjectFactory getObjectFactory();
+  IObjectFactory2 getObjectFactory2();
 
   /**
    * @return The output directory used for the reports.
    */
-  public String getOutputDirectory();
+  String getOutputDirectory();
 
   /**
    * @return true if the tests must be run in parallel.
    */
-  public String getParallel();
+  String getParallel();
 
-  public String getParentModule();
+  String getParentModule();
 
-  public String getGuiceStage();
+  String getGuiceStage();
 
   /**
    * @return The value of this parameter, or null if none was specified.
    */
-  public String getParameter(String parameterName);
+  String getParameter(String parameterName);
 
   /**
    * Retrieves the map of groups and their associated test methods.
    *
    * @return A map where the key is the group and the value is a list
    * of methods used by this group.
    */
-  public Map<String, Collection<ITestNGMethod>> getMethodsByGroups();
-
-  /**
-   * Retrieves the list of all the methods that were invoked during this run.
-   * @return a collection of ITestNGMethods belonging to all tests included in the suite.
-   * @deprecated Use getAllInvokedMethods().
-   */
-  @Deprecated
-  public Collection<ITestNGMethod> getInvokedMethods();

```

Comment:
@krmahadevan, please keep this api, as it's used by testng-remote:
https://github.com/testng-team/testng-remote/blob/master/remote/src/main/java/org/testng/remote/strprotocol/SuiteMessage.java#L40

Reply:
Oh Good lord.. looks like I opened up the pandora's box Nick :) 

btw, is it not possible for you to switch to using `getAllInvokedMethods()` instead ? But I guess it will break all those users who are not upgrading to the latest version of the eclipse plugin, but have a newer version of TestNG in their classpath right ? 

Btw, can you please do me a favor and quickly run through this PR and tell me what else is a problem with this PR @missedone